tensorflow/python/ops/special_math_ops.py

```diff
@@ -324,13 +324,43 @@ def sort_key(input_index, a):
     num_summed_elements = _total_size(t0_shape[-len(axes_to_sum):])
     new_shape = t0_shape[:len(preserved_axes)] + (num_broadcast_elements_t0,
                                                   num_summed_elements)
+    # Make new_shape valid, if invalid to be used in reshape
+    none_dims = [i for i, d in enumerate(new_shape)
```

Comment:
That would be the simplest way, however the reason I tried to avoid fixing like that way is that it might cause some performance overhead.
As far as I understood, if one can determine the shape by just using `get_shape`, then node creation caused by `array_ops.shape` does not occur, thus there exists a possibility to gain performance a bit.
I don't know how large the overhead of adding `array_ops.shape` node to the graph is. If it is negligible, then replacing `get_shape` with `array_ops.shape` would be the simplest and the cleanest way to fix this issue.